cmd/server-main.go

```diff
@@ -461,6 +464,35 @@ func initConfigSubsystem(ctx context.Context, newObject ObjectLayer) error {
 	return nil
 }
 
+func getGlobalInternodeInterface(interfs ...string) string {
+	globalInternodeInterfaceOnce.Do(func() {
+		if len(interfs) != 0 && strings.TrimSpace(interfs[0]) != "" {
+			globalInternodeInterface = interfs[0]
+			return
+		}
+		ip := "127.0.0.1"
+		host, _ := mustSplitHostPort(globalMinioAddr)
+		if host != "" {
+			ip = host
+		}
+
+		ifs, err := net.Interfaces()
+		logger.FatalIf(err, "Unable to get IP addresses of this host")
+
+		for _, interf := range ifs {
+			addrs, err := interf.Addrs()
+			if err == nil {
+				for _, addr := range addrs {
+					if strings.SplitN(addr.String(), "/", 2)[0] == ip {
+						globalInternodeInterface = interf.Name
+					}
+				}
+			}
+		}
```

Comment:
There is a non-zero chance that `globalInternodeInterface` ends up being `""` here.

Maybe set `globalInternodeInterface = ip` in that case.

Reply:
If the loop above doesn't match anything we can still end up with an empty string.

Just set `globalInternodeInterface = ip` unconditionally above, instead of depending on the error value.